### PR TITLE
cleanup(driver,userspace/libscap): merged syscall_code_routing_table with syscall_table

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area driver-kmod

/area driver-bpf

/area libscap

**Does this PR require a change in the driver versions?**

/version driver-API-version-major

Because it changes the maps idx for bpf driver

**What this PR does / why we need it**:

Added a `ppm_sc` field to syscall_table entries, and merged the same-idx routing table and syscall table.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
Please see original https://github.com/falcosecurity/libs/pull/571 self-review.

**Review by leogr — APPROVED** [review]

**Review by Andreagit97 — APPROVED** [review]
Left some comments not to be addressed here, just to remember them :) Amazing work @FedeDP :rocket: 
/approve

File: driver/bpf/maps.h [hunk 1]
```diff
@@ -34,13 +34,6 @@ struct bpf_map_def __bpf_section("maps") tail_map = {
 	.max_entries = PPM_FILLER_MAX,
 };
 
-struct bpf_map_def __bpf_section("maps") syscall_code_routing_table = {
```
[the remark this answers is not in the document]
Comment (Andreagit97) [reply driver/bpf/maps.h:37]: Addressed here https://github.com/falcosecurity/libs/pull/592

**Pull request merged**